Week3/homework/step4-bonus.js

```diff
@@ -1,10 +1,17 @@
 'use strict';
 
-const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
 
-// Add your function here. Try and come up with a good name for this function
 
-// Replace `yourFunction` with the name of the function you just created
-const uniqueValues = yourFunction(values);
+const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
+
 
-console.log(uniqueValues);
+let uniqueValues = function (array) {
+    let uniquearray = []
+    for (let i = 0; i < array.length; i++) {
+        if (uniquearray.indexOf(array[i]) === -1 && array[i] !== '') {
+            uniquearray.push(array[i]);
+        }
+    }
```

Comment:
Excellent.  Well done.

Reply:
Just so you know. There is a new datatype in Javascript called a Set (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) which only stores unique values. If the order of the resulting array does not matter, then you can do this, and have your unique values:
```
const uniques = Array.from(new Set(values));
```